xarray/core/computation.py

```diff
@@ -396,8 +399,37 @@ def _unpack_dict_tuples(
     return out
 
 
+def _check_core_dims(signature, variable_args, name):
+    """
+    Chcek if an arg has all the core dims required by the signature.
+
+    Slightly awkward design, of returning the error message. But we want to
+    give a detailed error message, which requires inspecting the variable in
+    the inner loop.
+    """
+    missing = []
+    for core_dims, variable_arg in zip(signature.input_core_dims, variable_args):
+        # Check whether all the dims are on the variable. Note that we need the
+        # `hasattr` to check for a dims property, to protect against the case where
+        # a numpy array is passed in.
+        if hasattr(variable_arg, "dims") and set(core_dims) - set(variable_arg.dims):
+            missing += [[variable_arg, core_dims]]
+    if missing:
+        message = f"Missing core dims from `{name}` on variable(s):\n\n"
```

Comment:
Now I'm a bit confused, is this wording correct/better?

```suggestion
        message = f"Missing core dims on some variables named `{name}`:\n\n"
```

Reply:
It is indeed...